charts/drax/charts/service-orchestrator/templates/configmap.yaml

```diff
@@ -10,6 +10,13 @@ top:
   {{ $ | toYaml | nindent 2 }}
 
 name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+{{- $config := .Values.config }}
+{{- $instance := include "accelleran.common.instance" (dict "top" .) }}
+
+{{- if and (hasKey $config "serviceMonitorUrl") (eq ($config.serviceMonitorUrl | default "") "") }}
+  {{- $config = merge $config (dict "serviceMonitorUrl" (printf "http://%s-service-monitor:80" $instance)) }}
+{{- end }}
```

Comment:
Maybe add this whole templated block which is pure variable assignment outside of the yaml/output (so above `top:`)